gsoc/models.py

```diff
@@ -0,0 +1,222 @@
+from django.db import models
+
+
+class Gsoc(models.Model):
+    count = models.IntegerField(primary_key=True)
+
+    def __str__(self):
+        return str(self.count)
+
+
+class Result(models.Model):
+    year = models.IntegerField(primary_key=True)
+    state = models.CharField(max_length=100)
+    is_current = models.BooleanField()
+    org_signup_open = models.BooleanField()
+    org_approval = models.BooleanField()
+    orgs_published = models.BooleanField()
+    student_signup_open = models.BooleanField()
+    slot_request = models.BooleanField()
+    slot_allocation_grace_period = models.BooleanField()
+    project_acceptance = models.BooleanField()
+    project_decisions_finalization = models.BooleanField()
+    first_work_period = models.BooleanField()
+    first_evaluations = models.BooleanField()
+    second_work_period = models.BooleanField()
+    second_evaluations = models.BooleanField()
+    third_work_period = models.BooleanField()
+    final_week = models.BooleanField()
+    final_evaluations_mentor = models.BooleanField()
+    post_program = models.BooleanField()
+    org_signup_open_starts = models.DateField()
```

Comment:
I'd prefer the dates are grouped into time periods.

This is a timeline of periods.

See https://developers.google.com/open-source/gsoc/timeline

"org_signup" should have a start and end date that are grouped together

"org_approval" should have a start and end date that are grouped together

etc

Reply:
@jayvdb,  I am not able to find a solution for how I group two dates together in Django model? Please help.

Reply:
@hemangsk @nkprince007 @mamat-rahmat , any ideas on how to merge these start & end dates.

Reply:
I think we need to use an existing django app or library to provide the data fields, and thus also hold the data.

https://djangopackages.org/grids/g/calendar/ shows the main options.

django-scheduler looks like it has the most features built in, while django-eventtools looks like the most raw & re-usable with fewer features.

Assuming django-eventtools is our best option, we would need a custom `GsocEvent(BaseEvent)` and `GsocOccurrence(BaseOccurrence)`, and the `Program` model would then need to have either
- an array of `GsocOccurrence` , and all of these start/end are pushed into it as events with the event name derived from the prefix (e.g. `org_signup_open` -> title="Org signup open")
- fields for each event, like `org_signup_open = GsocOccurrence()`

the later feels like the better option, at least initially.